rust/src/data_structures.rs

```diff
@@ -389,6 +477,93 @@ pub enum DtlsFingerprint {
     },
 }
 
+fn hex_as_bytes(input: &str, output: &mut [u8]) {
+    for (i, v) in output.iter_mut().enumerate() {
+        *v = u8::from_str_radix(&input[i * 3..(i * 3) + 2], 16)
+            .map_err(|error| {
+                panic!("Failed to parse value {input} as series of hex bytes: {error}")
+            })
+            .unwrap();
+    }
```

Comment:
Generally try to avoid manual indexing in loops in Rust, use higher level operations to get an iterator and zip it with the other iterator to move in lock step. This also typically results in vectorized machine code that is good for performance too.

Something like this:
```suggestion
    for (i, o) in input.splot(':').zip(&mut output) {
        *o = u8::from_str_radix(i, 16)
            .unwrap_or_else(|error| {
                panic!("Failed to parse value {i} as series of hex bytes: {error}")
            });
    }
```

Reply:
applied, thanks.